### PR TITLE
feat: Log task execution

### PR DESCRIPTION
**Pull request by SilviaZeta**
# Description
Add decorator for logging the start and the end of a task.

gliff-ai/roadmap/issues/54, snackbar to inform about a task in progress.

# Dependency changes
No

# Testing
No

# Documentation
No

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New migrations have been committed
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ChasNelson1990 — COMMENTED** [review]
So this just gives us "doing something" vs "done" - right? What about discovering "n out of N images uploaded" - can we do that?

**Comment by SilviaZeta** [comment]
> So this just gives us "doing something" vs "done" - right? What about discovering "n out of N images uploaded" - can we do that? (maybe a follow on PR?)

Yes, this only works when curate is loaded onto dominate and triggers a snackbar with spinner like this:
Screenshot 2021-07-09 at 14 45 54

Yes, it would be nice to have a progress bar! Next PR..

**Review by ChasNelson1990 — APPROVED** [review]

File: src/ui.tsx [hunk 1]
```diff
@@ -156,6 +158,9 @@ class UserInterface extends Component<Props, State> {
       thumbnailHeight: thumbnailSizes[2].size,
       selectMultipleImagesMode: false,
     };
+
+    /* eslint-disable @typescript-eslint/unbound-method, @typescript-eslint/no-unsafe-assignment */
+    this.addUploadedImage = this.addUploadedImage.bind(this);
```
Comment (cooper667) [inline src/ui.tsx:163]: Is there a way to make it work on methods without doing this? ie. keep the methods as arrow functions?

**Review by cooper667 — APPROVED** [review]

File: src/ui.tsx (hunk 1 above)
Comment (SilviaZeta) [reply src/ui.tsx:163]: Yes, I'd like to get rid of that, but I cannot find a way to get decorators to work with arrow functions.

**Pull request merged**